### ISSUE TITLE
confusing error if wrong extension

### ISSUE THREAD

**Issue by edublancas** [ploomber/ploomber#472]
ploomber infers the type of task with a combination of the source and product extension, but if there's a typo a confusing error shows. example, if the user wants to dump data:

```yaml
  
  - source: another.sql 
    product: another.parquets # typo here
```
error:

```pytb
ploomber.exceptions.SourceInitializationError: Error initializing SQLScriptSource('SELECT *\nFROM nu...ams"], "param") ]]') Loaded from: /Users/Edu/Desktop/sql/another.sql: The {{product}} placeholder is required. Example: 'CREATE TABLE {{product}} AS (SELECT * FROM ...)'
```

Fix: see if the value in `product` has something similar to a `.parquet` or `.csv` extension and show an error

**Comment by idomic** [comment]
@lgfunderburk another straightforward one :)

**Comment by idomic** [comment]
@jramirez857 Another good first one

**Comment by jramirez857** [comment]
Sure, I can take this up!

**Comment by neelasha23** [comment]
@edublancas Is this issue sill open? Can I take it up?

**Comment by idomic** [comment]
I think @jramirez857 already started working on this

**Comment by jramirez857** [comment]
@neelasha23 feel free to take this! I am currently working on a different issue.

**Comment by edublancas** [comment]
@neelasha23 as a pointer, you can use [this](https://docs.python.org/3/library/difflib.html#difflib.get_close_matches) to compute string distance to see if the user had a typo (e.g. parque instead of parquet)

**Comment by neelasha23** [comment]
Sure! Starting on this..

**Comment by neelasha23** [comment]
@edublancas 
I run the SQL template pipeline : https://github.com/ploomber/projects/tree/master/templates/spec-api-sql
After changing the extension like so:
```
- class: SQLDump
    source: join_dump.sql
    product: output/joined_data.cv
    name: join_dump
    chunksize: null
```
the pipeline still runs successfully and creates `output/joined_data.cv` which has proper data. 

The same thing happened when i changed .parquet to .parquets (https://github.com/ploomber/projects/tree/master/guides/first-pipeline)  ; in fact the pipeline still runs successfully with proper output data even if I remove the .parquet extension completely.

Am i missing something here?

**Comment by edublancas** [comment]
the example is passing `class: SQLDump`, which tells Ploomber the type of task that is, please remove that line:

```yaml
-  source: join_dump.sql
    product: output/joined_data.cv
    name: join_dump
    chunksize: null
```

then let me know if you are able to reproduce the error

**Comment by neelasha23** [comment]
@edublancas 
Yes I was able to reproduce an error on removing the task type line:
```
Loading pipeline...
Error: Error initializing SQLiteRelation with source: 'output/joined_data.cv'
SQLRelationPlaceholder must be initialized a collection of 2 or 3 elements, got: 'output/joined_data.cv' (21 elements). Example: ["schema", "name", "table"]
```

I havent been able to debug the code successfully as of yet. I tried following this guide: https://github.com/ploomber/projects/tree/master/guides/debugging
However the debug() didnt work for the SQL task. Is there any other guide I can refer to?

I have created a PR based on my understanding of the code to get some feedback: https://github.com/ploomber/ploomber/pull/674
